env_store.go

```diff
@@ -157,7 +158,10 @@ func getProviderPath(path string) (string, string) {
 		var fileProviderName = file.ProviderName
 		return fileProviderName, strings.TrimPrefix(path, "file:")
 	}
-	if strings.HasPrefix(path, "vault:") {
+	// If the path contains the "vault:" substring it is most probably a vault path
+	// otherwise the injector will not process it
+	re := regexp.MustCompile(`(vault:\w+)(\/\w+)`)
```

Comment:
Can we use a more generic regexp here such as `(vault:)(.*)#(.*)` to make it more consistent? This might not catch all the cases, while the false-positives will be handled by the internal package anway.